src/kubernetes/state/instrumenting.rs

```diff
@@ -96,73 +96,57 @@ mod tests {
         }
     }
 
-    fn get_metric_value(op_kind: &'static str) -> Option<metrics_runtime::Measurement> {
-        let controller = crate::metrics::CONTROLLER.get().unwrap_or_else(|| {
-            crate::metrics::init().unwrap();
-            crate::metrics::CONTROLLER
-                .get()
-                .expect("failed to init metric container")
+    fn get_metric_value(op_kind: &'static str) -> Option<MetricValue> {
+        let controller = crate::metrics::get_controller().unwrap_or_else(|_| {
+            crate::metrics::init().expect("unable to init metrics");
+            crate::metrics::get_controller().expect("failed to init metric container")
         });
```

Comment:
Would it make sense to roll this into `fn get_controller`?

Reply:
I thought about it, and it looks like it only makes sense for tests. Otherwise, to extend this idea further, we could use [`once_cell::sync::Lazy`](https://docs.rs/once_cell/1.4.1/once_cell/sync/struct.Lazy.html) for the initialization, and get rid of the error altogether.
However, my thinking is, in the usual application flow, we'd rather have more control on the initialization time of the metrics, rather than doing it lazily. The problem there is if we couple `get_controller` with `init` like that, then people might assume it's ok to just use `get_controller` and skip the `init` in a proper location. But `init` is required, because it also makes `metrics` macros work, and those can't be coupled with `init` internally. Thus, it's easier for us to reason about this whole system if we don't couple `get_controller` with `init`.

Actually, I probably should call `init` here unconditionally to avoid the confusion. It's implemented as an idempotent call, so it'll be ok.